libpromises/rlist.c

```diff
@@ -527,6 +542,17 @@ Rlist *RlistAppend(Rlist **start, const void *item, RvalType type)
         break;
 
     case RVAL_TYPE_LIST:
+        if (allow_all_types)
+        {
+            JsonElement* store = JsonArrayCreate(RlistLen(item));
+            for (const Rlist *rp = item; rp; rp = rp->next)
+            {
+                JsonArrayAppendElement(store, RvalToJson(rp->val));
+            }
+
+            return RlistAppendRval(start, (Rval) { store, RVAL_TYPE_CONTAINER });
+        }
+
```

Comment:
Ok, please make a comment that it behaves this way above the function definition.